src/m365/teams/commands/app/app-list.ts

```diff
@@ -30,59 +33,112 @@ class TeamsAppListCommand extends GraphItemsListCommand<TeamsApp> {
     const telemetryProps: any = super.getTelemetryProperties(args);
     telemetryProps.all = args.options.all || false;
     telemetryProps.teamId = typeof args.options.teamId !== 'undefined';
+    telemetryProps.teamName = typeof args.options.teamName !== 'undefined';
     return telemetryProps;
   }
 
-  public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
-    let endpoint: string = '';
+  private getTeamId(args: CommandArgs): Promise<string> {
     if (args.options.teamId) {
-      endpoint = `${this.resource}/v1.0/teams/${encodeURIComponent(args.options.teamId)}/installedApps?$expand=teamsApp`;
-
-      if (!args.options.all) {
-        endpoint += `&$filter=teamsApp/distributionMethod eq 'organization'`;
-      }
+      return Promise.resolve(args.options.teamId);
     }
-    else {
-      endpoint = `${this.resource}/v1.0/appCatalogs/teamsApps`;
 
-      if (!args.options.all) {
-        endpoint += `?$filter=distributionMethod eq 'organization'`;
-      }
-    }
+    const teamRequestOptions: any = {
+      url: `${this.resource}/v1.0/me/joinedTeams?$filter=displayName eq '${encodeURIComponent(args.options.teamName as string)}'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
 
-    this
-      .getAllItems(endpoint, logger, true)
-      .then((): void => {
-        if (args.options.output === 'json') {
-          logger.log(this.items);
+    return request
+      .get<{ value: Team[] }>(teamRequestOptions)
+      .then(response => {
+        const teamItem: Team | undefined = response.value[0];
+
+        if (!teamItem) {
+          return Promise.reject(`The specified team does not exist in the Microsoft Teams`);
         }
-        else {
-          if (args.options.teamId) {
-            logger.log((this.items as unknown as TeamsAppInstallation[]).map(i => {
-              return {
-                id: i.id,
-                displayName: i.teamsApp.displayName,
-                distributionMethod: i.teamsApp.distributionMethod
-              };
-            }));
-          }
-          else {
-            logger.log(this.items.map(i => {
-              return {
-                id: i.id,
-                displayName: i.displayName,
-                distributionMethod: i.distributionMethod
-              };
-            }));
-          }
+
+        if (response.value.length > 1) {
+          return Promise.reject(`Multiple Microsoft Teams teams with name ${args.options.teamName} found: ${response.value.map(x => x.id)}`);
         }
 
-        if (this.verbose) {
-          logger.log(chalk.green('DONE'));
+        return Promise.resolve(teamItem.id);
+      });
+  }
+
+  private getEndPointURL(args: CommandArgs): Promise<string> {
+    let endpoint: string = '';
```

Comment:
Since we're not using the endpoint anywhere outside of the Promise, there is no need to define it as a variable here.